SU-SWS/stanford_courses
# Courses: Mapping a tag to something that is human-readable

# READY FOR REVIEW

@jbickar, this one is a doozy.

Steps to test:

1. Install a JSA website somewhere
2. Check out this branch
3. Enable stanford_courses_tag_translate
4. Create a course importer using http://explorecourses.stanford.edu/search?view=xml-20130201&filter-coursestatus-Active=on&page=0&catalog=&academicYear=&q=IR%3A%3Awim&collapse=
5. Find a course that was imported and review the tags that are attached
6. Go to: `admin/config/stanford/courses/tag-translate` and create a few new tag translations
7. Go back to importer and re-import
8. Go back to the course you reviewed earlier and check the human readable field for the new values.
9. Go to `admin/config/stanford/courses/tag-translate/export` and keep that information handy

### Part II: Import / Export

1. Install a new JSA website somewhere else
2. Check out this branch and enable stanford_courses_tag_translate
3. Go to `admin/config/stanford/courses/tag-translate/import` and import the values from the steps above.
4. Go to `admin/config/stanford/courses/tag-translate` and verify they imported correctly.

### Part III: Additional checks.

1. Check that the content type has the human readable field
2. Check that the feed importer has a mapping to the human readable field
3. Check that the feeds tamper has the new custom plugin on the human readable field
4. Try to import bad values.
5. Disable and uninstall to ensure that everything goes away correctly.
6. Before enabling the stanford_courses_tag_translate modify the feeds importer, feeds tampers, and content type. 

### Part IV: More things to check
1. Set the default value for the import to [original] and run an import
2. Set the default value for the import to nothing (empty field) and run an import
3. Set the default value for the import to "Some random string" and run import
4. Check to see that `admin/structure/views/view/stanford_courses_ref_human_tag` installs on enable
5. Check to see that `admin/structure/views/view/stanford_courses_ref_human_tag` uninstalls on uninstall
6. Check that Deprecated field group is on courses with two tag fields in them.

Thanks,
